Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+    $frontDoorsEndpointWithoutReqMinTLSVersion = @()
+
+    $frontDoorsEndpointWithReqMinTLSVersion = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
+
+    $FrontDoorEndpoints = @()
+
+    $NonCompliantFrontDoorEndpoints = @()
+    $ResourceAppIdURI = "https://management.azure.com/"
+    $customDomainResponse =@()
+    $AccessMgtToken= (Get-AzAccessToken -ResourceUrl $ResourceAppIdURI).Token
+    
+    Write-Host "[Step 3 of 4] Fetch all Front Door configurations"
+    Write-Host $([Constants]::SingleDashLine)
+
+        $FrontDoorResources | ForEach-Object {
+                $frontDoorResource = $_
+                $resourceGroupName = $_.ResourceGroupName
+                $resourceName = $_.Name
+                $frontDoorEp = [FrontendEndPointsClass]:: new()
+                try
+                {
+                    Write-Host "Fetching Front Door Endpoint configuration: Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+                    Write-Host $([Constants]::SingleDashLine)
+                
+                    $FrontDoorEndpoints = Get-AzFrontDoorCdnCustomDomain -ProfileName $resourceName -ResourceGroupName $resourceGroupName
+                
+                    if($FrontDoorEndpoints)
+                    {
+                        Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+                        Write-Host $([Constants]::SingleDashLine)
+                    }
+
+                    if($null -ne $FrontDoorEndpoints)
+                    {
+                        foreach ($item in $FrontDoorEndpoints) 
+                        {
+                            $domainName = $item.Name
+                            $domainResourceGroupName = $item.ResourceGroupName
+                            $TypeOfCertificate= @()
+                            $MinTLSVersion = @()
+                    
+                            if($null -ne $AccessMgtToken)
+                            {
+                                $header = "Bearer " + $AccessMgtToken
+                                $headers = @{"Authorization"=$header;"Content-Type"="application/json"; "x-ms-version" ="2013-08-01"}
+                                $uri = [string]:: Format("{0}/subscriptions/{1}/resourceGroups/{2}/providers/Microsoft.Cdn/profiles/{3}/customDomains/{4}?api-version=2021-06-01",$ResourceAppIdURI,$SubscriptionId,$domainResourceGroupName,$resourceName,$domainName)
+                                $customDomainResponse = Invoke-WebRequest -Method GET -Uri $uri -Headers $headers -UseBasicParsing
+                    
+                                if($customDomainResponse.StatusCode -ge 200 -and $customDomainResponse.StatusCode -le 399)
+                                {
```

Comment:
What will happen if we don't get response code that we are expecting?

Reply:
Added logic if response is not generated.